parsec/core/types/backend_address.py

```diff
@@ -78,7 +84,24 @@ def from_url(cls, url: str):
         else:
             params = {}
 
-        path = unquote_plus(split.path)
+        if allow_http_redirection and split.scheme in ("http", "https"):
+            # `no_ssl` is defined by http/https scheme and shouldn't be
+            # overwritten by the query part of the url
+            params["no_ssl"] = ["true" if split.scheme == "http" else "false"]
```

Comment:
Also, why are parameters a list of string instead of a simple string?

Reply:
it's because an url param can be repeated multiple times (e.g. `http://example.com/foo?spam=1&spam=2`, which `parse_qs` would turn into `{"spam": [1, 2]}`)